doc/Overview/Mvux/Advanced/Commands.md

```diff
@@ -0,0 +1,209 @@
+
+A command is an object that implements the [`ICommand`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.input.icommand) interface. This interface has two methods:
+
+- [`Execute`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.input.icommand.execute): This method is called when the command is executed.
+- [`CanExecute`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.input.icommand.canexecute): This method is invoked when the UI needs to determine if the command can be executed. It returns a boolean value indicating if the command can be executed.
+
+Commands can be bound to UI elements. To do this, the `Command` property of the UI element can be used. For example, the following code binds the `Click` event of a button to the `Save` command:
+
+```
+<Button Command="Save">Save</Button>
+```
+
+## Asynchronous commands
+
+The MVUX [`IAsyncCommand`](https://github.com/unoplatform/uno.extensions/blob/main/src/Uno.Extensions.Reactive/Presentation/Commands/IAsyncCommand.cs) interface, is a Command that implements [`ICommand`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.input.icommand?view=windows-app-sdk-1.3) and adds support for asynchronous operations.  
+As it implements `ICommand`, it can be bound from the View as a regular Command (e.g. in a `Button.Command` property). The advantage is that it can be invoked asynchronously.  
+In addition, it also implements [`INotifyPropertyChanged`](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.inotifypropertychanged), to enable tracking if its properties have changed, and [`ILoadable`](xref:Toolkit.Controls.LoadingView#iloadable) - an Uno interface that provides information of an object's state whether it's currently in execution mode or not, and notifies subscribers when this state changes.
+
+![A class diagram of System.ComponentModel.ICommand inheritence structure](../Assets/Commands-2.jpg)
+
+## Creating commands
+
+There are several methods of how to create an MVUX command.
+
+1. Using code generation
+
+    ### Basic commands
+
+    When creating a method in the Model, it will be generated as an `AsyncCommand` in the Bindable Model.
+
+    For example, when declaring a method of the following signatures:
+
+    ```csharp
+    public void DoWork();
+    public ValueTask DoWork();
+    public ValueTask DoWork(CancellationToken ct);    
+    ```
+
+    A `DoWork` command will be generated in the Bindable Proxy Model. When that command is executed via a button-click etc., the method in the Model will be called:
+
+    ```xml
+    <Button Command="{Binding DoWork}" />
+    ```
+
+    As you may have noticed, the method can be asynchronous and take a `CancellationToken`, but these are not mandatory and commands will be generated for synchronous methods as well.
```

Comment:
"As you may have noticed" is very vague.
Is it important that the reader has noticed?

This also suggests that the three `DoWork` methods defined above are showing possible options, rather than examples of three methods that can/should be implemented in the described scenario.
Making it clearer above that _any_ of the above method signatures can be used would mean this paragraph is not needed.